In [52]:
import kaggle

# Envoi du fichier final et récupération du résultat

In [53]:
competition = 'house-prices-advanced-regression-techniques'

In [54]:
submission_file = "test_predict.csv"

In [55]:
kaggle.api.competition_submit(submission_file, "01-R-randomforest-with-mssub-qual", competition)

100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 33.1k/33.1k [00:02<00:00, 12.2kB/s]


Successfully submitted to House Prices: Advanced Regression Techniques

In [58]:
kaggle.api.competitions_submissions_list(competition)

[{'date': '2019-12-01T17:01:38.14Z',
  'description': '01-R-randomforest-with-mssub-qual',
  'errorDescription': None,
  'fileName': 'test_predict.csv',
  'privateScore': None,
  'publicScore': '0.15256',
  'ref': 13572231,
  'status': 'complete',
  'submittedBy': 'laurent33',
  'submittedByRef': 'laurent33',
  'teamName': 'laurent33',
  'totalBytes': 33912,
  'type': 'standard',
  'url': 'https://www.kaggle.com/submissions/13572231/13572231.raw'},
 {'date': '2019-12-01T16:47:36Z',
  'description': '01-R-randomforest-with-corr>0.7kept',
  'errorDescription': None,
  'fileName': 'test_predict.csv',
  'privateScore': None,
  'publicScore': '0.15181',
  'ref': 13572096,
  'status': 'complete',
  'submittedBy': 'laurent33',
  'submittedByRef': 'laurent33',
  'teamName': 'laurent33',
  'totalBytes': 33943,
  'type': 'standard',
  'url': 'https://www.kaggle.com/submissions/13572096/13572096.raw'},
 {'date': '2019-12-01T16:38:48.077Z',
  'description': '01-R-randomforest-with-corr<0.5',
  'er

# Historiques scores house-prices (top score = 0.07)

- score de 0.177 :
    - préparation des données sur 100% du Train puis split
    - traitement des données manquantes sur Train = suppression des lignes
    - modèle entraîné sur 80% Trainfull (score obtenu de 0.138 sur set de validation 20% Train)
    - traitement des données manquantes sur Test = mise à 0 si quantitative, 'None' si qualitative
        - Pb1 : préparation fait sur 100% du trainfull au lieu de se restreindre au train <br>
        - Pb2 : traitements différents sur données d'entraînement et données de test (l'on ne peut pas supprimer d'input sur ce dernier car il faut faire une prédiction)
        - Pb3 : variables considées à tort comme quantitatives (MSSubClass)

- score de 0.168 => **amélioré**
    - préparation des données sur 100% du Train puis split
    - traitement des données manquantes sur Train = **mise à 0 si quantitative, 'None' si qualitative**
    - modèle entraîné sur 80% Trainfull (score obtenu de **0.152** sur set de validation 20% Train)
    
        - Pb1 : préparation fait sur 100% du trainfull au lieu de se restreindre au train <br>
        - Pb3 : variables considées à tort comme quantitatives (MSSubClass)

- score de 0.168 => *pas de changement*
    - préparation des données sur 100% du Train puis split
    - traitement des données manquantes sur Train = mise à la **médiane** si quantitative, 'None' si qualitative
    - modèle entraîné sur 80% Trainfull (score obtenu de **0.155** sur set de validation 20% Train)
        - Pb1 : préparation fait sur 100% du trainfull au lieu de se restreindre au train <br>
        - Pb3 : variables considées à tort comme quantitatives (MSSubClass)

- score de 0.182 => *moins bon*
    - préparation des données sur 100% du Train puis split
    - traitement des données manquantes sur Train = mise à la médiane si quantitative, 'None' si qualitative
    - suppression de 7 features corrélées > **0.5** (au lieu 0.7 précédemment)
    - modèle entraîné sur 80% Trainfull (score obtenu de **0.18** sur set de validation 20% Train)
        - Pb1 : préparation fait sur 100% du trainfull au lieu de se restreindre au train <br>
        - Pb3 : variables considées à tort comme quantitatives (MSSubClass)

- score de 0.152 => **amélioré**
    - préparation des données sur 100% du Train puis split
    - traitement des données manquantes sur Train = mise à la médiane si quantitative, 'None' si qualitative
    - les 5 features corrélées > 0.7 **non supprimées**
    - modèle entraîné sur 80% Trainfull (score obtenu de **0.14** sur set de validation 20% Train)
        - Pb1 : préparation fait sur 100% du trainfull au lieu de se restreindre au train <br>
        - Pb3 : variables considées à tort comme quantitatives (MSSubClass)

- score de 0.153 => pas d'amélioration
    - préparation des données sur 100% du Train puis split
    - traitement des données manquantes sur Train = mise à la médiane si quantitative, 'None' si qualitative
    - les 5 features corrélées > 0.7 non supprimées
    - MSSubClass convertie en variable **qualitative**
    - modèle entraîné sur 80% Trainfull (score obtenu de **0.15** sur set de validation 20% Train)
        - Pb1 : préparation fait sur 100% du trainfull au lieu de se restreindre au train <br>